packages/lit-dev-content/src/components/algolia-search-controller.ts

```diff
@@ -63,10 +75,27 @@ export class AgloliaSearchController<T extends {}> {
     if (trimmedQuery.length < 2) {
       return [];
     }
-    const results = await this._index.search<T>(trimmedQuery, {
+    type SearchOptionsReadonly = NonNullable<
+      Parameters<typeof this._index.search>[1]
+    >;
+    type SearchOptions = Writeable<
+      SearchOptionsReadonly,
+      keyof SearchOptionsReadonly
+    >;
```

Comment:
I am not following the types and why they are required? Is it for the `searchOpts.attributesToHighlight` mutation happening beneath?

Would it be possible to pass `this._attributesToHighlight` explicitly in the search options literal?

I.e.:

```
{ page: 0, hitsPerPage: 10, attributesToHighlight: this._attributesToHighlight }
```

Although that assumes an explicit `undefined` value is treated the same as not providing the option at all.

Reply:
set some sensible defaults